impacket/smbserver.py

```diff
@@ -319,246 +329,262 @@ def queryFsInformation(path, filename, level=0, pktFlags = smb.SMB.FLAGS2_UNICOD
         fileAttributes = attribs
         return fileSize, lastWriteTime, fileAttributes
 
-def findFirst2(path, fileName, level, searchAttributes, pktFlags = smb.SMB.FLAGS2_UNICODE, isSMB2 = False):
-     # TODO: Depending on the level, this could be done much simpler
-     
-     #print "FindFirs2 path:%s, filename:%s" % (path, fileName)
-     fileName = os.path.normpath(fileName.replace('\\','/'))
-     # Let's choose the right encoding depending on the request
-     if pktFlags & smb.SMB.FLAGS2_UNICODE:
-         encoding = 'utf-16le'
-     else:
-         encoding = 'ascii'
-
-     if len(fileName) > 0 and (fileName[0] == '/' or fileName[0] == '\\'):
+
+def findFirst2(path, fileName, level, searchAttributes, pktFlags=smb.SMB.FLAGS2_UNICODE, isSMB2=False):
+    # TODO: Depending on the level, this could be done much simpler
+
+    # print "FindFirs2 path:%s, filename:%s" % (path, fileName)
+    fileName = os.path.normpath(fileName.replace('\\', '/'))
+    # Let's choose the right encoding depending on the request
+    if pktFlags & smb.SMB.FLAGS2_UNICODE:
+        encoding = 'utf-16le'
+    else:
+        encoding = 'ascii'
+
+    if len(fileName) > 0 and (fileName[0] == '/' or fileName[0] == '\\'):
         # strip leading '/'
         fileName = fileName[1:]
 
-     pathName = os.path.join(path,fileName)
-     files = []
-
-     if pathName.find('*') == -1 and pathName.find('?') == -1:
-         # No search patterns
-         pattern = ''
-     else:
-         pattern = os.path.basename(pathName)
-         dirName = os.path.dirname(pathName)
-
-     # Always add . and .. Not that important for Windows, but Samba whines if 
-     # not present (for * search only)
-     if pattern == '*':
-         files.append(os.path.join(dirName,'.'))
-         files.append(os.path.join(dirName,'..'))
-
-     if pattern != '':
-         for file in os.listdir(dirName):
-             if fnmatch.fnmatch(file.lower(),pattern.lower()):
+    if not isInFileJail(path, fileName):
+        LOG.error("Path not in current working directory")
+        return [], 0, STATUS_NOT_SUPPORTED
```

Comment:
Is `STATUS_NOT_SUPPORTED` the right error code here?

Reply:
To answer all 3 questions, I wanted to kill the session in the case Path Traversal is attempted. in each def I tried to see what the most appropriate error code is (same with the STATUS_NOT_SUPPORTED in the last def), but if there is a more appropriate error code for each case I will be happy to change it.